### DISCUSSION TITLE
Question before I start

### DISCUSSION THREAD

**Discussion by camj256** [collin80/SavvyCAN#360] [Q&A]
First off, awesome project. I just want to check and see if something is possible or probable. I've got a 2011 Escalade I swapped to a standalone ECU. The trunk won't open no matter what and we think its because the car doesn't realize its in park without the ECU/TCM attached. So we think (but aren't sure) that the neutral safety switch runs over the GMLAN once it hits the ecm/tcm and that its just a safety so i don't accidently open the trunk going down the highway. If anyone knows if it actually runs over the GMLAN off the top of their head that would be outstanding. I don't suspect anyone will. I'm happy to put in the time but I just want to know its going to be possible sniff out the "i'm in park" signal for another truck and then write some arduino software to just replay that on the network so I can open this stupid trunk again. I am a software engineer with a computer engineering background so I'm capable if its possible. I just don't want to waste a ton of time if someone can read this and tell me go for it or not in 10 seconds. Thanks!!!

**Comment by collin80** [comment]
It sounds entirely possible that you've properly identified the problem. And, it's entirely possible to sniff the traffic on another vehicle and write an arduino sketch to emulate the proper signal. Most certainly you will find that gear position is sent over CAN bus. Is your vehicle checking that before allowing you to open the trunk? It's possible. It certainly could do so.

There exist some open files for gmlan that you might find useful. For instance, here is the link to a DBC file where you will find a current gear readout: https://github.com/commaai/opendbc/blob/master/gm_global_a_powertrain.dbc

It's labeled ECMPRDNL and has ID 309 (decimal) / 0x135
